go/src/meguca/server/router.go

```diff
@@ -248,15 +248,55 @@ func healthCheck(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte("God's in His heaven, all's right with the world"))
 }
 
-// Get HookTube title from ID
-func getHookTubeTitle(w http.ResponseWriter, r *http.Request) {
-	s, err := goscraper.Scrape(
-		"https://hooktube.com/embed/"+extractParam(r, "id"), 3)
+// Get YouTube title and googlevideo URL from URL
+func youTubeData(w http.ResponseWriter, r *http.Request) {
+	info, err := ytdl.GetVideoInfo("https://www.youtube.com/watch?v=" + extractParam(r, "id"))
 
 	if err != nil {
 		httpError(w, r, err)
 		return
 	}
 
-	w.Write([]byte(s.Preview.Title))
+	vidFormats := info.Formats.
+		Filter(ytdl.FormatExtensionKey, []interface{}{"webm"}).
+		Filter(ytdl.FormatResolutionKey, []interface{}{"360p"})
+	
+	if len(vidFormats) == 0 {
+		vidFormats = info.Formats.
+			Filter(ytdl.FormatExtensionKey, []interface{}{"webm"}).
+			Worst(ytdl.FormatResolutionKey)
+		
+		if len(vidFormats) == 0 {
+			httpError(w, r, err)
+			return
+		}
+	}
+
+	video, err := info.GetDownloadURL(vidFormats[0])
+
+	if err != nil {
+		httpError(w, r, err)
+		return
+	}
+
+	vidHighFormats := info.Formats.
+		Filter(ytdl.FormatExtensionKey, []interface{}{"webm"}).
+		Best(ytdl.FormatResolutionKey)
+	
+	if len(vidHighFormats) == 0 {
+		httpError(w, r, err)
+		return
+	}
+	
+	videoHigh, err := info.GetDownloadURL(vidHighFormats[0])
+
+	if err != nil {
+		httpError(w, r, err)
+		return
+	}
+
+	w.Write([]byte(
+		info.Title + "\n" +
+		info.GetThumbnailURL(ytdl.ThumbnailQualityDefault).String() + "\n" +
+		video.String() + "\n" + videoHigh.String()))
```

Comment:
This will perform 7+ extra heap allocations for no reason. Go strings are immutable and a new one will be allocated on each concatenation. Please use fmt.Fprintf() instead.